src/couch_replicator_httpc_pool.erl

```diff
@@ -31,8 +31,7 @@
 -record(state, {
     url,
     limit,                  % max # of workers allowed
-    free = [],              % free workers (connections)
-    busy = [],              % busy workers (connections)
+    conns = [],
```

Comment:
Oh nice. Makes things simpler here. At first seems awkward to have a connection pool which also gets items from a connection pool. But it seems this connection pool is used to limit concurrency per replication, which is a useful thing.